src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -1748,26 +1748,22 @@ private async Task WriteStreamPropertiesAsync(SelectExpandNode selectExpandNode,
                     changedProperties = deltaObject.GetChangedPropertyNames();
                 }
 
-                foreach (KeyValuePair<IEdmStructuralProperty, PathSelectItem> complexProperty in complexProperties)
+                if (changedProperties != null)
                 {
-                    if (changedProperties == null || changedProperties.Contains(complexProperty.Key.Name))
+                    IDictionary<IEdmStructuralProperty, PathSelectItem> newDic = new Dictionary<IEdmStructuralProperty, PathSelectItem>();
```

Comment:
maybe name this more appropriately to make it easier for readers of the code to know what it is for, not just that it is a dictionary.
```suggestion
                    IDictionary<IEdmStructuralProperty, PathSelectItem> propertiesToReturn = new Dictionary<IEdmStructuralProperty, PathSelectItem>();
```

Reply:
#Resolved.